qutebrowser/config/configdata.yml

```diff
@@ -1251,10 +1251,15 @@ tabs.padding:
   type: Padding
   desc: Padding (in pixels) around text for tabs.
 
-tabs.persist_mode_on_change:
-  default: false
-  type: Bool
-  desc: Stay in insert/passthrough mode when switching tabs.
+tabs.mode_on_change:
+  default: normal
+  type:
+    name: String
+    valid_values:
+      - persist: "Retain the current mode."
+      - restore: "Restore previously saved mode."
+      - normal:  "Always revert to normal mode."
+  desc: When switching tabs, what input mode is applied.
```

Comment:
This will need some migration code so old `autoconfig.yml` files which set `tabs.persist_mode_on_change` don't break.

I've now cleaned up the related code a bit in `master`, so can you please merge the latest `master` (or rebase your branch on the latest `master`) and add a migration?

To do so, you should edit `qutebrowser/config/configfiles.py`. In `YamlConfig._handle_migrations`, after the `for` loop which handles simple removed/renamed options, you should be able to do:

```python3
persist = 'tabs.persist_mode_on_change'
if persist in self._values:
    if self._values[persist]:
        self._values['tabs.mode_on_change'] = 'persist'
    else:
        self._values['tabs.mode_on_change'] = 'normal'
    del self._values['tabs.persist_mode_on_change']
```

Please also add a corresponding test to `tests/unit/config/test_configfiles.py` which makes sure the value gets changed correctly.

If you need help, as usual, ask in the IRC channel :wink: